tests/framework/helpers/BaseConsoleTest.php

```
@@ -86,5 +94,130 @@ public function testAnsiStrlen(): void
+        $expected = BaseConsole::renderColoredString('%rст%gОк%n');
+        $this->assertSame($expected, $actual);
+
+        $str = BaseConsole::renderColoredString('%rТест%gABC%n');
+        $actual = BaseConsole::ansiColorizedSubstr($str, 4, 2);
+        $expected = BaseConsole::renderColoredString('%gAB%n');
+        $this->assertSame($expected, $actual);
+    }
+
+    public function testMarkdownToAnsi(): void
+    {
+        $result = BaseConsole::markdownToAnsi('**bold** text');
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\yii\helpers\BaseConsole' in method 'testMarkdownToAnsi'. (undefined)

(StaticAccess)